cpp/include/cuspatial/experimental/point_polygon_distance.cuh

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cuspatial {
+
+/**
+ * @ingroup distance
+ * @brief Computes pairwise multipoint to multipolygon distance
+ *
+ * @tparam MultiPointRange An instance of template type `MultiPointRange`
+ * @tparam MultiPolygonRange An instance of template type `MultiPolygonRange`
+ * @tparam OutputIt iterator type for output array. Must meet the requirements of [LRAI](LinkLRAI).
+ *
+ * @param multipoints Range of multipoints, one per computed distance pair.
+ * @param multipolygons Range of multilinestrings, one per computed distance pair.
+ * @return Output Iterator past the last distance computed
+ *
+ * [LinkLRAI]: https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator
+ * "LegacyRandomAccessIterator"
+ */
+template <class MultiPointRange, class MultiPolygonRange, class OutputIt>
+OutputIt pairwise_point_polygon_distance(MultiPointRange multipoints,
+                                         MultiPolygonRange multipoiygons,
+                                         OutputIt distances_first,
+                                         rmm::cuda_stream_view stream = rmm::cuda_stream_default);
```

Comment:
Params do not match the docs. Returning an `OutputIt` and also taking one as undocumented argument seems fishy.

Reply:
It's standard for most of our algorithms, just like STL. Returns an iterator to one past the last computed results, as the @return doc says.

I think the doc for `@tparam OutputIt` needs more detail to say that this must contain a floating point type.

Reply:
> must contain a floating point type.

More precisely, "must be convertible from floating point type", since user can use output transform iterator to convert the result and transform into other types.